### ISSUE TITLE
Message metadata, to allow detection of encryption type of a message.

### ISSUE THREAD

**Issue by mofosyne** [ircv3/ircv3-specifications#113]
Wonder if there is a defined way for an IRC client to read metadata attached to a room, individual messages or individuals, that would make secure chat addons easier to interface with. If there is, then you can disregard this request.

It may be a tall order to ask all IRC clients to support a standard way to secure messages besides SSL. But for those who want more advance stuff like OTR, or public/private style encryption. At the very least, attaching user/room/message definable metadata fields may help.

This would allow for addons programs to existing IRC clients, to identify which message can be read by it. Maybe you could also define a field in individual user metadata for people to place their preferred public key. Etc...

Basically, the goal is to make IRC extensible for secure communication, via consistent metadata for addon programs, rather than mandating better encryption.

**Comment by grawity** [comment]
For _message_ metadata, I'd say use v3.1 message tags and write a spec that allows specific tags – for example, prefixed `user.*` or `client/*` – to be sent from one client to another. (The default is to discard unknown tags, so that a client could not spoof e.g. `time` tags.)

For users and channels, there's a v3.2 metadata specification; see if that could be used.

**Comment by DarthGandalf** [comment]
message tags are 3.2 too

2015-02-21 9:42 GMT-08:00 Mantas Mikulėnas <notifications@github.com>:

**Comment by Aerdan** [comment]
`metadata-3.2` will not support channels at this time, since it depends on `MONITOR` which is standardization of an existing mechanism that only supports nicks.

**Comment by jwheare** [comment]
This is covered by message tags and metadata.
